File: src/main/java/com/mindgate/service/BookingRequestService.java
package com.mindgate.service;

import java.util.List;

import com.mindgate.pojo.BookingRequest;

public interface BookingRequestService {
	public boolean addNewBookingRequest(BookingRequest bookingRequestId);

	public boolean updateBookingRequest(BookingRequest bookingRequestId);

	public List<BookingRequest> getAllBookingRequest();

	public BookingRequest getBookingRequest(int bookingRequestId);

	public List<BookingRequest> getBookingRequestByEmployeeId(int employeeId);

	public boolean deleteBookingRequest(int bookingRequestId);

	public BookingRequest getRequest(int bookingRequestId);

	public List<BookingRequest> getBookingRequest();
	
	public List<BookingRequest> getBookingRequestToDirector();
	
	public List<BookingRequest> getBookingRequestForDirector();

	
	public BookingRequest getRequestDirector(int bookingRequestId);
	
	public List<BookingRequest> getResubmitDocuments(int employeeId);
	
	


}

File: app/booking-request/booking-request.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { BookingRequest } from './booking-request/booking-request';

@Injectable({
  providedIn: 'root'
})
export class BookingRequestService {
  private baseURL = 'http://localhost:8080/bookingRequest/';
  constructor( private http: HttpClient) { }

  sendBookingRequest(bookingRequest: BookingRequest) : Observable<any>
  {
    console.log("in sendBookingRequest" + bookingRequest);
    
    return this.http.post(this.baseURL+'newBookingRequest',bookingRequest);

  }

  updateBookingRequest(bookingRequest: BookingRequest) : Observable<any>
  {
    console.log("in sendBookingRequest");
    return this.http.put(this.baseURL+'updateBookingRequest/',bookingRequest);
    
    }

  getBookingRequestByEmployeeId(employeeId: number) : Observable<any>
  {
    return this.http.get(`${this.baseURL+'employee/'}${employeeId}`);
  }

  getBookingRequest(bookingRequestId: number) : Observable<any>
  {
    console.log("in getBookingRequest service");
    
    return this.http.get(this.baseURL+bookingRequestId);
  }
  
   getAllBookingRequest() : Observable<any>
  {
    return this.http.get(this.baseURL+'allBookingRequests');
  }

  getResubmitDocuments(employeeId: number): Observable<any>
  {
    return this.http.get(this.baseURL+'/resubmit/'+employeeId);
  }


}


